File: LeetCode/739. Daily Temperatures.py
### KNOWLEDGE IS POWER  ###

class Solution(object):
    def dailyTemperatures(self, T):
        """
        :type temperatures: List[int]
        :rtype: List[int]
        """
        ans, s = [0]*len(T), deque()
        for cur, cur_tmp in enumerate(T):
            while s and cur_tmp > T[s[-1]]:
                ans[s[-1]] = cur - s[-1]
                s.pop()
            s.append(cur)
        return ans

File: LeetCode/739. Daily Temperatures.cpp
/***  KNOWLEDGE IS POWER  ***/

class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temperatures) {
        int n = temperatures.size();
        vector<int>v(n, 0);
        stack<int>st;
        
        for(int i = n-1; i>=0; i--)
        {
            while(!st.empty() && temperatures[st.top()] <= temperatures[i])
                st.pop();
           if(!st.empty())
                v[i] = st.top()-i; 
            st.push(i);
        }
        
        return v;

    }
};
